lnwire/channel_update_2.go

```diff
@@ -0,0 +1,510 @@
+		var htlcMin = uint64(c.HTLCMinimumMsat)
+		records = append(records, tlv.MakePrimitiveRecord(
+			ChanUpdate2HTLCMinMsatType, &htlcMin,
+		))
+	}
+
+	var htlcMax = uint64(c.HTLCMaximumMsat)
+	records = append(records, tlv.MakePrimitiveRecord(
+		ChanUpdate2HTLCMaxMsatType, &htlcMax,
+	))
+
+	if c.FeeBaseMsat != defaultFeeBaseMsat {
+		records = append(records, tlv.MakePrimitiveRecord(
+			ChanUpdate2FeeBaseMsatType, &c.FeeBaseMsat,
+		))
+	}
+
+	if c.FeeProportionalMillionths != defaultFeeProportionalMillionths {
+		records = append(records, tlv.MakePrimitiveRecord(
+			ChanUpdate2FeeProportionalMillionthsType,
+			&c.FeeProportionalMillionths,
+		))
+	}
+
+	err := EncodeMessageExtraDataFromRecords(&c.ExtraOpaqueData, records...)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.ExtraOpaqueData, nil
+}
+
+// MsgType returns the integer uniquely identifying this message type on the
+// wire.
+//
+// This is part of the lnwire.Message interface.
+func (c *ChannelUpdate2) MsgType() MessageType {
+	return MsgChannelUpdate2
+}
+
+// A compile time check to ensure ChannelUpdate2 implements the
+// lnwire.Message interface.
+var _ Message = (*ChannelUpdate2)(nil)
+
+// SCID returns the ShortChannelID of the channel that the update applies to.
+//
+// NOTE: this is part of the ChannelUpdate interface.
+func (c *ChannelUpdate2) SCID() ShortChannelID {
+	return c.ShortChannelID
+}
+
+// IsNode1 is true if the update was produced by node 1 of the channel peers.
+// Node 1 is the node with the lexicographically smaller public key.
+//
+// NOTE: this is part of the ChannelUpdate interface.
+func (c *ChannelUpdate2) IsNode1() bool {
+	return !c.Direction
+}
+
+// IsDisabled is true if the update is announcing that the channel should be
+// considered disabled.
+//
+// NOTE: this is part of the ChannelUpdate interface.
+func (c *ChannelUpdate2) IsDisabled() bool {
+	return !c.DisabledFlags.IsEnabled()
+}
+
+// GetChainHash returns the hash of the chain that the message is referring to.
+//
+// NOTE: this is part of the ChannelUpdate interface.
+func (c *ChannelUpdate2) GetChainHash() chainhash.Hash {
+	return c.ChainHash
+}
+
+// ForwardingPolicy returns the set of forwarding constraints of the update.
+//
+// NOTE: this is part of the ChannelUpdate interface.
+func (c *ChannelUpdate2) ForwardingPolicy() *ForwardingPolicy {
+	return &ForwardingPolicy{
+		TimeLockDelta: c.CLTVExpiryDelta,
+		BaseFee:       MilliSatoshi(c.FeeBaseMsat),
+		FeeRate:       MilliSatoshi(c.FeeProportionalMillionths),
+		MinHTLC:       c.HTLCMinimumMsat,
+		HasMaxHTLC:    true,
+		MaxHTLC:       c.HTLCMaximumMsat,
+	}
+}
+
+// CmpAge can be used to determine if the update is older or newer than the
+// passed update. It returns 1 if this update is newer, -1 if it is older, and
+// 0 if they are the same age.
+//
+// NOTE: this is part of the ChannelUpdate interface.
+func (c *ChannelUpdate2) CmpAge(update ChannelUpdate) (int, error) {
+	other, ok := update.(*ChannelUpdate2)
```

Comment:
Assuming `BlockHeight` is part of the interface, then we won't need to do a type cast here? This is basically the same as the `CmpAgge` on the other one, maybe we can: 
```go
func CmpAge[U ChannelUpdate1 | ChannelUpdate2](update1 U, update 2 U)
```

?

Dunno if would work out tho, I think you need to do a soft cast to be able to access teh shareed fields still